main.go

```diff
@@ -217,6 +333,44 @@ func advertiseFeatureLabels(helper APIHelpers, labels Labels) error {
 	return nil
 }
 
+// advertiseResources advertises the discovered resources to a Kubernetes node
+// via the API server.
+func advertiseResources(helper APIHelpers, resources api.ResourceList) error {
+	cli, err := helper.GetClient()
+	if err != nil {
+		stderrLogger.Printf("can't get kubernetes client: %s", err.Error())
+		return err
+	}
+	// Get the current node.
+	node, err := helper.GetNode(cli)
+	if err != nil {
+		stderrLogger.Printf("failed to get node: %s", err.Error())
+		return err
+	}
+	patch := []map[string]string{}
+	for name, quantity := range resources {
+		patch = append(patch, map[string]string{
+			"op":    "add",
+			"path":  fmt.Sprintf("/status/capacity/%s", escapeForJSONPatch(name)),
+			"value": quantity.String(),
+		})
+	}
+	patchJSON, err := json.Marshal(patch)
+	if err != nil {
+		stderrLogger.Printf("could not prepare patch to advertise opaque resources: %s", err)
+	}
+	return cli.Core().RESTClient().Patch(apitypes.JSONPatchType).Resource("nodes").Name(node.Name).SubResource("status").Body(patchJSON).Do().Error()
```

Comment:
Question: what happens when patchJSON is empty (i.e., no integer resources were discovered)?